jbmc/src/java_bytecode/java_utils.cpp

```diff
@@ -510,3 +511,41 @@ void set_declaring_class(symbolt &symbol, const irep_idt &declaring_class)
 {
   symbol.type.set(ID_C_class, declaring_class);
 }
+
+static const symbolt *find_cprover_nondet_initialize_on_class(
+  const symbol_tablet &symbol_table,
+  const std::string &class_name)
+{
+  const irep_idt method_name(class_name + ".cproverNondetInitialize:()V");
+  const auto ret = symbol_table.symbols.find(method_name);
+  if(ret != symbol_table.symbols.end())
+  {
+    return &ret->second;
+  }
+  return nullptr;
+}
+
+const symbolt *find_cprover_nondet_initialize(
+  const symbol_tablet &symbol_table,
+  std::string class_name)
+{
+  namespacet ns{symbol_table};
+  const symbolt *ret =
+    find_cprover_nondet_initialize_on_class(symbol_table, class_name);
+
+  while(!ret)
+  {
+    const irep_idt &class_id =
+      to_java_class_type(ns.lookup(class_name).type).get_super_class();
```

Comment:
Worth being more defensive here? I can't remember if `ns.lookup` throws on not found or returns a default.

Reply:
`ns.lookup` has an INVARIANT that the thing was found. That's the only reason I'm using a namespace instead of the symbol table directly, to avoid having to handle that error myself.

`to_java_class_type` asserts the type has ID_struct. If it doesn't have a super class then `get_super_class()` will return an empty irep_idt, which is handled on the next line.

So actually I think the errors are handled okay, despite how little explicit error handling there is.